schema.json

```diff
@@ -0,0 +1,488 @@
+                                    "type": "boolean",
+                                    "description": "Optional property, to indicate if a struct member is optional"
+                                }
+                            },
+                            "required": [
+                                "name",
+                                "doc",
+                                "type"
+                            ]
+                        }
+                    }
+                },
+                "required": [
+                    "name",
+                    "doc",
+                    "type"
+                ]
+            }
+        },
+        "functions": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "description": "Name of the function"
+                    },
+                    "doc": {
+                        "type": "string"
+                    },
+                    "returns": {
+                        "type": "object",
+                        "description": "Optional property, describing the return type of the function",
+                        "properties": {
+                            "doc": {
+                                "type": "string"
+                            },
+                            "type": {
+                                "type": "string",
+                                "description": "Return type of the function"
+                            }
+                        },
+                        "required": [
+                            "doc",
+                            "type"
+                        ]
+                    },
+                    "args": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "type": "string",
+                                    "description": "Name of the function argument"
+                                },
+                                "doc": {
+                                    "type": "string"
+                                },
+                                "type": {
+                                    "type": "string",
+                                    "description": "Argument type"
+                                },
+                                "pointer": {
+                                    "type": "string",
+                                    "description": "Optional property, specifies if a function argument type is a pointer",
+                                    "enum": [
+                                        "immutable",
+                                        "mutable"
+                                    ]
```

Comment:
(Can be future work)
This is used a bunch of times and I think could be defined by `$ref` as well (link again: https://stackoverflow.com/a/62585173/1268131)

`functions.items.properties.args` and `objects.items.properties.methods.items.properties.args` also could probably be the same, and maybe `structs.items.properties.members` too depending if it can be made to match (dawn does this too)